stl/inc/ranges

```diff
@@ -776,6 +777,63 @@ namespace ranges {
         };
     };
 
+    template <class _Fn, class... _Types>
+    class _Range_closure : public _Pipe::_Base<_Range_closure<_Fn, _Types...>> {
+    public:
+        _STL_INTERNAL_STATIC_ASSERT((same_as<decay_t<_Types>, _Types> && ...));
+        _STL_INTERNAL_STATIC_ASSERT(is_empty_v<_Fn>&& is_default_constructible_v<_Fn>);
+
+        template <class... _UTypes>
+        constexpr explicit _Range_closure(_UTypes&&... _Args) noexcept(
+            (is_nothrow_constructible_v<_Types, _UTypes> && ...))

```

Comment:
It couldn't hurt. Also, I think the `_STL_INTERNAL_STATIC_ASSERT` should be a constraint to avoid this template ever pretending to be a default/copy/move constructor.